### PR TITLE
gracefully kill processes. dont send crazy segments along. add window…

### PR DESCRIPTION
**Pull request by kevinkreiser**
…ing based on inactivity

### CODE REVIEW

File: py/simple_reporter.py [hunk 1]
```diff
@@ -80,6 +77,12 @@ def split(l, n):
     pos = end
   return result
 
+def interrupt_wrapper(func):
```
Comment (kevinkreiser) [inline py/simple_reporter.py:80]: lets us gracefully kill sub processes

**Comment by heffergm** [comment]
👍

File: py/simple_reporter.py [hunk 2]
```diff
@@ -135,54 +140,75 @@ def match(file_names, config, quantisation, source, dest_dir):
...
-          s = [
-            str(r['id']), str(r.get('next_id', INVALID_SEGMENT_ID)), str(duration), '1',
-            str(r['length']), str(r['queue_length']), str(start), str(end), source, 'AUTO'
-          ]
-          tiles.setdefault(tile_file_name, []).append(','.join(s) + os.linesep)
+
+      #find the points where inactivity occurs, these are the window boundaries
+      windows = []
+      for i, point in enumerate(all_points):
+        if i == 0 or point['time'] - points[i - 1]['time'] > inactivity:
```
Comment (kevinkreiser) [inline py/simple_reporter.py:153]: here we find where there was inactivity in the trace and mark the index. we then only match windows where there arent large gaps of inactivity

File: py/simple_reporter.py [hunk 3]
```diff
@@ -135,54 +140,75 @@ def match(file_names, config, quantisation, source, dest_dir):
...
+
+      #for each window, the last one is just an end marker
+      for i, window in enumerate(windows, len(windows) - 1):
+        #skip short traces
+        j = windows[i + 1]
+        if j - i < 2:
+          continue
+        
+        #get the matches for it
+        points = all_points[i : j]
```
Comment (kevinkreiser) [inline py/simple_reporter.py:165]: from here down the code is the same just indented

File: py/simple_reporter.py [hunk 4]
```diff
@@ -135,54 +140,75 @@ def match(file_names, config, quantisation, source, dest_dir):
...
+        buckets = (points[-1]['time'] - points[0]['time']) / quantisation + 1
+        segments = [ r for r in report['datastore']['reports'] if r['t0'] > 0 and r['t1'] > 0 and r['t1'] - r['t0'] > .5 and r['length'] > 0 and r['queue_length'] >= 0 ]
+        for r in segments:
+          duration = int(round(r['t1'] - r['t0']))
+          start = int(math.floor(r['t0']))
+          end = int(math.ceil(r['t1']))
+          min_bucket = int(start / quantisation)
+          max_bucket = int(end / quantisation)
+          diff = max_bucket - min_bucke
+          if diff > buckets:
```
Comment (kevinkreiser) [inline py/simple_reporter.py:187]: except for this line where we actually throw out segments that have crazy times on them. this happens very very rarely but for some reason the start and end times on certain traces' segments have times that are either way in the past or way in the future so we toss those and will fix that problem upstream

**Pull request merged**